experimental/FTheoryTools/src/G4Fluxes/special_attributes.jl

```diff
@@ -638,3 +640,347 @@ function well_quantized_and_vertical_ambient_space_models_of_g4_fluxes(m::Abstra
+false
+
+julia> c = [60, 51, 90, 0, 24, 51, -24, 45, 30, 0, -48, 90, -57, 60, 30, 15, 120, 0, -60, 0, -720, -420, -270, -60, -2190];
+
+julia> qsm_g4_candidate = g4_flux(qsm_model, sum(c[i]*g4_classes[i] for i in 1:length(g4_classes)), check = false)
+G4-flux candidate
+  - Elementary quantization checks: not executed
+  - Tadpole cancellation check: not executed
+  - Verticality checks: not executed
+  - Non-Abelian gauge group: breaking pattern not analyzed
+
+julia> passes_elementary_quantization_checks(qsm_g4_candidate)
+true
+
+julia> passes_verticality_checks(qsm_g4_candidate)
+true
+
+julia> breaks_non_abelian_gauge_group(qsm_g4_candidate)
+false
+```
+"""
+function well_quantized_and_vertical_and_no_non_abelian_gauge_group_breaking_ambient_space_models_of_g4_fluxes(m::AbstractFTheoryModel; check::Bool = true)::Tuple{QQMatrix, QQMatrix}
+
+  # (1) Entry checks
+  @req base_space(m) isa NormalToricVariety "Computation of well-quantized G4-fluxes only supported for toric base and ambient spaces"
+  @req dim(ambient_space(m)) == 5 "Computation of well-quantized G4-fluxes only supported for 5-dimensional toric ambient spaces"
+  if check
+    @req is_complete(ambient_space(m)) "Computation of well-quantized G4-fluxes only supported for complete toric ambient spaces"
+    @req is_simplicial(ambient_space(m)) "Computation of well-quantized G4-fluxes only supported for simplicial toric ambient space"
+  end
+  if has_attribute(m, :well_quantized_and_vertical_and_no_non_abelian_gauge_group_breaking_ambient_space_models_of_g4_fluxes)
+    return get_attribute(m, :well_quantized_and_vertical_and_no_non_abelian_gauge_group_breaking_ambient_space_models_of_g4_fluxes)
+  end
+
+
+  # (2) Compute data, that is frequently used by the sophisticated intersection product below
+  S = cox_ring(ambient_space(m))
+  gS = gens(cox_ring(ambient_space(m)))
+  linear_relations = matrix(QQ, matrix(ZZ, rays(ambient_space(m))))
+  scalings = [c.coeff for c in S.d]
+  mnf = Oscar._minimal_nonfaces(ambient_space(m))
+  sr_ideal_pos = [Vector{Int}(Polymake.row(mnf, i)) for i in 1:Polymake.nrows(mnf)]
+  data = (
+    S = S,
+    gS = gS,
+    linear_relations = linear_relations,
+    scalings = scalings,
+    sr_ideal_pos = sr_ideal_pos
+  )
+
+
+  # (3) Are intersection numbers known?
+  inter_dict = Dict{NTuple{4, Int64}, ZZRingElem}()
+  s_inter_dict = Dict{String, ZZRingElem}()
+  if has_attribute(m, :inter_dict)
+    inter_dict = get_attribute(m, :inter_dict)
+  end
+  if has_attribute(m, :s_inter_dict)
+    s_inter_dict = get_attribute(m, :s_inter_dict)
+  end
+
+
+  # (4) Obtain critical information - this may take significant time!
+  ambient_space_flux_candidates_basis = ambient_space_models_of_g4_fluxes(m, check = check)
+  list_of_base_divisor_pairs_to_be_considered = Oscar._ambient_space_base_divisor_pairs_to_be_considered(m)
+  ambient_space_flux_candidates_basis_indices = get_attribute(m, :ambient_space_models_of_g4_fluxes_indices)
```

Comment:
```suggestion
  ambient_space_flux_candidates_basis_indices = get_attribute(m, :ambient_space_models_of_g4_fluxes_indices)::SomeType
```
if you know this type, please put it here